### PR TITLE
Functions

### PR DESCRIPTION
**Pull request by mrniranjan**
Example programs on  IO, Switch, string and defer .

### CODE REVIEW

File: basics/file_os_create.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	"os"
+)
+
+func main() {
+	f, err := os.Create("a.txt")
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+	l, err := f.WriteString("Hello World")
```
Comment (gsr-shanks) [inline basics/file_os_create.go]: Not sure, but is it ok to redeclare the same variable?

File: basics/file_os_create.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	if err != nil {
+		fmt.Println(err)
+		f.Close()
+	}
+	fmt.Println(l, "bytes written successfully")
+	err = f.Close()
+
```
Comment (gsr-shanks) [inline basics/file_os_create.go]: Empty line

File: basics/file_os_create.go [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+func main() {
+	f, err := os.Create("a.txt")
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+	l, err := f.WriteString("Hello World")
+	if err != nil {
+		fmt.Println(err)
+		f.Close()
```
Comment (gsr-shanks) [inline basics/file_os_create.go]: IDE is pointing out 'Unhandled error' for f.Close()
Do we need to close() here?

File: basics/file_os_create.go (hunk 2 above)
Comment (mrniranjan) [reply basics/file_os_create.go]: Fixed it in the new commit

File: basics/file_os_create.go (hunk 1 above)
Comment (mrniranjan) [reply basics/file_os_create.go]: fixed it in the new commit.

File: basics/file_os_create.go (hunk 3 above)
Comment (mrniranjan) [reply basics/file_os_create.go]: fixed it in the new commit.

**Comment by mrniranjan** [comment]
@gsr-shanks hey can you have a look at the PR.

**Pull request merged**